docs/development/pr-review.asciidoc

```diff
@@ -0,0 +1,129 @@
+
+There may also be times when a person is inspired by a particular contribution to introduce a new way to style code that we already have different style guidelines or "prior art" for. It's OK to bring this up in a pull request, but ultimately that discussion should branch off into a separate issue or pull request to update the appropriate guide. If this change is prompted by a reviewer, then the original PR should not be blocked on this.  If the change is prompted by the author, then they can either update the PR to be consistent with our existing guidelines (preferred) or they can choose to block the PR entirely on that separate styleguide discussion.
+
+
+[float]
+=== Nitpicking
+
+Nitpicking is when a reviewer identifies trivial and unimportant details in a pull request and asks the author to change them. This is a completely subjective category that is impossible to define universally, and it's equally impractical to define a blanket policy on nitpicking that everyone will be happy with.
+
+Reviewers should feel comfortable giving any feedback they have on a pull request regardless of how trivial it is. Authors should feel equally comfortable passing on feedback that they think is trivial and inconsequential.
+
+Often, reviewers have an opinion about whether the feedback they are about to give is a nitpick or not. While not required, it can be really helpful to identify that feedback as such, for example "nit: a newline after this would be helpful". This helps the author understand your intention.
+
+
+[float]
+=== Review feedback that is never appropriate
+
+Whether or not a bit of feedback is appropriate for a pull request is really dependent on the motivation for giving the feedback in the first place. Reviewers should not be reviewing code with the mindset of "how would I write this code?". The reviewer didn't write the code, and their goal isn't to craft the contribution into a form that is simply whatever they would have written if they had.
+
+Feedback such as "this is crap" isn't feedback at all. It's both mean and unhelpful, and it is never appropriate.
+
+
+[float]
+=== A checklist
+
+Establishing a comprehensive checklist for all of the things that should happen in all possible pull requests is impractical, but that doesn't mean we lack a concrete set of minimum requirements that we can enumerate. The following items should be double checked for any pull request:
+
+* CLA check passes
+* Jenkins job runs and passes
+* Adheres to the spirit of our various styleguides
+* Has thorough unit test coverage
+* Automated tests provide high confidence the change continues to work without manual verification
+* Appropriate product documentation is included (asciidocs)
+* Any new UI changes are accessible to differently abled persons, including but not limited to sufficient contrasts in colors, keyboard navigation, and aria tags
+* Includes APIs for new or changed functionality, either programmatically for plugins or as REST endpoints
```

Comment:
This isn't described in the text above, I think(?) I'd remove it for now

Reply:
I don't think it's a problem that this checklist includes items that aren't explicitly called out above. But I'll remove this for the sake of getting this initial set of guidelines in, since this hasn't been a long established practice or anything like that.

Reply:
Removed